### PR TITLE
Pull in 3-class models for review

### LINKED ISSUE #78: Create more rigorous train/val/test sets out of the new labels - separate by image-date instead of random assignment

### PR DESCRIPTION
**Pull request by steeleb**
Hey Caitlin! Long time no PR. Thanks for still being willing to play the Superior PR game. 

Trying to keep this pretty chill with a lot of redundant code across missions, per usual. 

This PR creates train/val splits for model development (script`08...`) and then builds the gradient tree boost models (with default settings) from the training data, saves performance statistics, and exports the raster to Drive (scripts `09...`-`13...`). 

For this PR, like those previous, make sure there aren't any copy-paste flubs (where I don't consistently edit mission names), and make sure the general logic makes sense (and matches the code!)

Thank you!

This closes #78 in script `modeling/08_Train_Test_Split_3class.Rmd` - I've implemented this type of train-validation split in the 5-class model, and will for future models, too (not in this PR, in future PRs).

This is progress on #79 (I'll have another PR that runs through the first half of the script for a 5-class model, and sounds like there is some interest in some other iterations that would show up in another PR (including combining LS5/7 and LS8/9 since the sensors are similar and atmospheric correction methodology is the same for those pairs).

### CODE REVIEW

File: modeling/11_Landsat_8_GTB_3class.Rmd [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+aoi_ee = ee.FeatureCollection('projects/ee-ross-superior/assets/aoi/Superior_AOI_modeling')
+
+def clip(image):
+  return image.clip(aoi_ee.geometry())
+
+
+date_length_8 = len(uniqueMissDate_l8.getInfo())
+
+# export tif to drive
+for d in range(63, date_length_8):
```
Comment (ccmothes) [inline modeling/11_Landsat_8_GTB_3class.Rmd:312]: What's the reason for '63' here?

**Review by ccmothes — APPROVED** [review]
Looks good! Just a few small mismatches I noticed and commented on but you might have reasons for those.

File: modeling/09_Landsat_5_GTB_3class.Rmd [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+    ])
+    
+```
+
+## Export GeoTiffs to drive
+
+Load task checker!
+
+```{python}
+##Function for limiting the max number of tasks sent to
```
[the remark this answers is not in the document]
Comment (steeleb) [reply modeling/09_Landsat_5_GTB_3class.Rmd:363]: It's SO USEFUL

File: modeling/08_Train_Test_Split_3class.Rmd [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+```
+
+
+## Build the train/validate sets
+
+Define the input features and output labels
+
+```{python}
+ls57_input_feat = ["SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B7"]
+ls89_input_feat = ["SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B8", "SR_B7"]
```
[the remark this answers is not in the document]
Comment (steeleb) [reply modeling/08_Train_Test_Split_3class.Rmd:404]: It sure should be.

File: modeling/11_Landsat_8_GTB_3class.Rmd (hunk 1 above)
Comment (steeleb) [reply modeling/11_Landsat_8_GTB_3class.Rmd:312]: Ha! This is because my loop stopped at index 63, so I had to restart it there.

Comment (steeleb) [inline modeling/11_Landsat_8_GTB_3class.Rmd:312]:
```suggestion
for d in range(date_length_8):
```

File: modeling/08_Train_Test_Split_3class.Rmd (hunk 3 above)
Comment (steeleb) [inline modeling/08_Train_Test_Split_3class.Rmd:404]:
```suggestion
ls89_input_feat = ["SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", "SR_B7"]
```

**Comment by steeleb** [comment]
THANK YOU!!! Fixed those little suckers, and merging (after a million years). I've got another one coming at you shortly.
